build/parseSpec.c

```diff
@@ -171,20 +171,64 @@ static int restoreFirstChar(rpmSpec spec)
     return 0;
 }
 
+static const struct lineTypes {
+    int id;
+    size_t textLen;
+    const char *text;
+    int withArgs;
+} specLineTypes[] = {
+    { LINE_IF,          LEN_AND_STR("%if")     , 1},
+    { LINE_IFARCH,      LEN_AND_STR("%ifarch") , 1},
+    { LINE_IFNARCH,     LEN_AND_STR("%ifnarch"), 1},
+    { LINE_IFOS,        LEN_AND_STR("%ifos")   , 1},
+    { LINE_IFNOS,       LEN_AND_STR("%ifnos")  , 1},
+    { LINE_ELSE,        LEN_AND_STR("%else")   , 0},
+    { LINE_ENDIF,       LEN_AND_STR("%endif")  , 0},
+    { LINE_INCLUDE,     LEN_AND_STR("%include"), 1},
+    { LINE_OTHER,       0, NULL                }
+};
+static const size_t numLineTypes = sizeof(specLineTypes)/sizeof(*specLineTypes);
+
+static int parseLineType(char *line)
+{
+    for (int i=0; i<(numLineTypes-1); i++) {
+	if (specLineTypes[i].withArgs) {
+	    if ( ISMACROWITHARG(line, specLineTypes[i].text, specLineTypes[i].textLen) )
+		return i;
+	} else {
+	    if ( ISMACRO(line, specLineTypes[i].text, specLineTypes[i].textLen) )
+		return i;
+	}
+    }
+
+    return numLineTypes-1;
+}
+
 static int expandMacrosInSpecBuf(rpmSpec spec, int strip)
 {
     char *lbuf = NULL;
+    char *auxBuf;
```

Comment:
Like mentioned in an earlier review, this should be a const pointer since it's not used to modify the contents.
It's also unnecessarily in the outer scope when it's only used inside the if below.
Finally, *buf* is a weird and misleading name for a temporary pointer to some other string. "s" is commonly used for such pointers in codebase, similarly to how "i" is idiomatically used for indexing a for-loop.